geniza/corpus/ja.py

```diff
@@ -80,28 +80,31 @@ def arabic_to_ja(text):
 
 
 def arabic_or_ja(text, boost=True):
-    """Convert text to arabic or judaeo-arabic string; boost arabic by default"""
-    # if there is no arabic text, return as is
-    if not contains_arabic(text):
-        return text
+    # any arabic phrase or word
+    sr = re.compile(r'"[ \u0600-\u06FF]+"|[\u0600-\u06FF]+')
 
-    # extract arabic words from the search query
-    arabic_words = re_AR_letters.findall(text)
-    # generate judaeo-arabic equivalents
-    ja_words = [arabic_to_ja(word) for word in arabic_words]
-    # iterate over the original and converted words together and combine
+    # find matches
+    arabic = sr.findall(text)
 
-    # add boosting so arabic matches will be more relevant,
-    # unless boosting is disabled
-    boost = "^2.0" if boost else ""
+    # find what surrounds matches
+    nonarabic = sr.split(text)
 
-    for i, arabic_word in enumerate(arabic_words):
-        ja_word = ja_words[i]
-        # if the words differ, combine them as an OR
-        # then replace them in the original search query
-        # (preserving any existing search syntax like quotes, wildcards, etc)
-        if arabic_word != ja_word:
-            ar_or_ja_word = "(%s%s|%s)" % (arabic_word, boost, ja_word)
-            text = text.replace(arabic_word, ar_or_ja_word)
+    # assert the logical
+    assert len(nonarabic) == len(arabic) + 1  # given split/findall
 
-    return text
+    # stitch back together
+    boost = "^2.0" if boost else ""
+    tokens = []
+    for i in range(len(nonarabic)):
+        nonarabic_word = nonarabic[i]
```

Comment:
cleaner to use enumerate
```suggestion
    for i, nonarabic_word in enumerate(nonarabic):
```